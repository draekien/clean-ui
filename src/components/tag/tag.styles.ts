import { alpha } from '@theme-ui/color';
import { ThemeUIStyleObject } from '@theme-ui/css';
import { Theme } from '../..';

export const tagContainerCss: ThemeUIStyleObject = {
  paddingY: 'xxs',
  backgroundColor: 'background-lighter',
  display: 'inline-flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  borderRadius: 'xs',
};

export const tagCss: ThemeUIStyleObject = {
  paddingX: 'xs',
};

export const closeButtonCss: ThemeUIStyleObject = {
  background: 'none',
  outline: 'none',
  border: 'none',
  paddingLeft: 'xxs',
  borderLeft: '1px solid',
  borderLeftColor: (t: Theme) => alpha('b-400', 0.25)(t),
  cursor: 'pointer',
  color: 'text',
  transition: 'all 100ms',
  ':hover': {
    color: 'error-300',
  },
  ':focus': {
    color: 'error-300',
  },
  ':active': {
    color: 'error-400',
  },
};
