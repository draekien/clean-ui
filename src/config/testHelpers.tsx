import React, { ReactNode } from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../components/theme/theme.provider';

export const themedRender = (element: ReactNode) =>
  render(<ThemeProvider>{element}</ThemeProvider>);
