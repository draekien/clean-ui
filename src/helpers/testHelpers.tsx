import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../components/theme/theme.provider';

export const themedRender = (element: React.ReactNode) => {
  return render(<ThemeProvider>{element}</ThemeProvider>);
};

export const hexToRgb = (hex: string) => {
  hex = hex.replace(/[^0-9A-F]/gi, '');
  const bigint = parseInt(hex, 16);
  const r = (bigint >> 16) & 255;
  const g = (bigint >> 8) & 255;
  const b = bigint & 255;

  return `rgb(${r}, ${g}, ${b})`;
};
