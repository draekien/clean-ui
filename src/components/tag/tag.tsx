/** @jsxImportSource theme-ui */
import React from 'react';
import { Icon } from '../icon/icon';
import { Text } from '../text/text';
import { tagCss, tagContainerCss, closeButtonCss } from './tag.styles';

export interface TagProps {
  value: string;
  onClose?: (value: string) => void;
}

export const Tag: React.FC<TagProps> = ({ value, onClose }) => {
  const handleClose = () => {
    onClose && onClose(value);
  };

  return (
    <span sx={tagContainerCss}>
      <span sx={tagCss}>
        <Text variant="eyebrow">{value}</Text>
      </span>
      {!!onClose && (
        <button sx={closeButtonCss} onClick={handleClose}>
          <Icon name="close" size="small" />
        </button>
      )}
    </span>
  );
};
