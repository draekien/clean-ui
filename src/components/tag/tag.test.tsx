import React from 'react';
import { Tag } from './tag';
import { themedRender } from '../../helpers/testHelpers';

describe('Tag component', () => {
  it('should match snapshot', () => {
    const { container } = themedRender(<Tag value="tag" onClose={() => {}} />);

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot when onClose is undefined', () => {
    const { container } = themedRender(<Tag value="tag" />);

    expect(container).toMatchSnapshot();
  });
});
