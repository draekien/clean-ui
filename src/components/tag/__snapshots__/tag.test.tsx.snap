// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component should match snapshot 1`] = `
<div>
  <span
    class="css-x2q8nv-Tag"
  >
    <span
      class="css-15nj5vr-Tag"
    >
      <div
        class="css-1v9wm6q-Text css-vurnku"
      >
        tag
      </div>
    </span>
    <button
      class="css-7iz1i8-Tag"
    >
      <i
        class="material-icons  css-y0j37o-Icon"
      >
        close
      </i>
    </button>
  </span>
</div>
`;

exports[`Tag component should match snapshot when onClose is undefined 1`] = `
<div>
  <span
    class="css-x2q8nv-Tag"
  >
    <span
      class="css-15nj5vr-Tag"
    >
      <div
        class="css-1v9wm6q-Text css-vurnku"
      >
        tag
      </div>
    </span>
  </span>
</div>
`;
